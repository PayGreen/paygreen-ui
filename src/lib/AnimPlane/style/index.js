import styled from 'styled-components';
import {
    planeAnimationDuration,
    dotsDelay,
    dotsTransitions,
    activeStyle,
} from './base';

const AnimPlaneBase = styled.div`
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;

    .icon {
        position: absolute;
        transform-origin: center;
        transform: scale(0);
        transition: all ${props => props.theme.transition.sm};
    }

    .illu {
        width: 100%;
        max-width: ${props => props.theme.imageSize.sm};
        fill: ${props => props.theme.color[props.colorTheme].main};

        .plane {
            transition: opacity ${props => props.theme.transition.sm};
            transition-delay: ${props => props.dotsCount * dotsDelay}s;
            transform-origin: center;
            animation-fill-mode: forwards;
            animation-delay: ${props => props.dotsCount * dotsDelay}s;
            animation-duration: ${planeAnimationDuration}s;
            animation-timing-function: linear;

            .normal {
                opacity: 0.4;
            }

            .light {
                opacity: 0.2;
            }
        }

        .dots {
            opacity: 0.4;

            path {
                opacity: ${props => (props.isActive ? 0 : 1)};
                transition: all 0;
                ${props => dotsTransitions(props.dotsCount)};
            }
        }
    }

    ${props => (props.isActive ? activeStyle : null)};
`;

export { AnimPlaneBase };
