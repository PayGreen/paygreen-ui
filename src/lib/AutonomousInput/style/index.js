import styled from 'styled-components';

const AutonomousInputBase = styled.div`

`;

export { AutonomousInputBase };