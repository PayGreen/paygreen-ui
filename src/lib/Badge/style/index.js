import styled from 'styled-components';
import { math } from 'polished';
import { badgeSize, originalStyle, reverseStyle } from './base';

const BadgeBase = styled.span`
    display: flex;
    justify-content: center;
    align-items: center;
    width: fit-content;
    border-radius: ${props => props.theme.radius.sm};
    height: ${props => math(props.theme.font.size[props.textSize] + ' + 1rem')};
    font-size: ${props => props.theme.font.size[props.textSize]};
    text-align: center;
    text-transform: uppercase;
    letter-spacing: ${props => props.theme.font.spacing};
    font-weight: ${props => props.theme.font.weight.bold};

    ${props => badgeSize[props.textSize]};
    ${props => (props.colorType === 'original' ? originalStyle : reverseStyle)};
`;

export { BadgeBase };
