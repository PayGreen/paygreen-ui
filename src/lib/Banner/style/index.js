import styled from 'styled-components';
import { waveOptions } from '../../../shared/constants';
import { color, waveScale } from './constants';

const BannerBase = styled.div`
    position: relative;
    margin-top: ${props => props.topStyle !== waveOptions.none ? props.theme.wave.double : 0};
    margin-bottom: ${props => props.bottomStyle !== waveOptions.none ? props.theme.wave.double : 0};
    padding: .1px 0;
    background-image: linear-gradient(
        ${props => color.first[props.gradient]},
        ${props => color.second[props.gradient]}
    );

    .waves {
        position: absolute;
        left: 0;

        .light {
            opacity: .5;
        }

        &:first-child {
            fill: ${props => color.first[props.gradient]};
            bottom: 100%;
            transform: scale(${props => waveScale[props.topStyle]}, ${waveScale.top});
            margin-bottom: -1px;
        }
        
        &:last-child {
            fill: ${props => color.second[props.gradient]};
            top: 100%;
            transform: scale(${props => waveScale[props.bottomStyle]}, ${waveScale.bottom});
            margin-top: -1px;
        }
    }
`;

export { BannerBase };