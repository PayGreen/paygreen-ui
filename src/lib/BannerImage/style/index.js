import styled from 'styled-components';
import { decorationOptions } from '../../../shared/constants';
import { topStyle, bottomStyle } from './base';

const BannerImageBase = styled.div`
    ${props => topStyle[props.topStyle]};
    ${props => bottomStyle[props.bottomStyle]};
    position: relative;
    min-height: ${({theme}) => theme.bannerHeight.sm};
    padding: 0 .1px ${({theme}) => theme.blockSpace.md} .1px;
    padding-top: ${({topStyle, theme}) => topStyle !== decorationOptions.none ? theme.wave.simple : 0};
    padding-bottom: ${({bottomStyle, theme}) => bottomStyle !== decorationOptions.none ? theme.wave.simple : 0};

    &::before,
    &::after {
        content: '';
        position: absolute;
        z-index: ${({theme}) => theme.zindex.background};
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        max-height: ${({theme}) => theme.bannerHeight.lg};
        mask-image: linear-gradient(to top, transparent, black ${({theme}) => theme.bannerHeight.sm});
    }

    &::before {
        background-image: url('${({image}) => image}');
        background-size: cover;
        filter: grayscale(1);
        opacity: .5;
    }

    &::after {
        background-color: ${({useColorTheme, theme, colorTheme}) => useColorTheme ? theme.color[colorTheme].light : null};
        opacity: .8;
    }
`;

export { BannerImageBase };