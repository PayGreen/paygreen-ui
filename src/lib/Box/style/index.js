import styled from 'styled-components';

const BoxBase = styled.div`
    max-width: 100%;

    padding-top: ${props => props.theme.space[props.paddingTop]};
    padding-right: ${props => props.theme.space[props.paddingRight]};
    padding-left: ${props => props.theme.space[props.paddingLeft]};
    padding-bottom: ${props => props.theme.space[props.paddingBottom]};
    
    margin-top: ${props => props.theme.space[props.marginTop]};
    margin-right: ${props => props.theme.space[props.marginRight]};
    margin-left: ${props => props.theme.space[props.marginLeft]};
    margin-bottom: ${props => props.theme.space[props.marginBottom]};
`;

export { BoxBase };
