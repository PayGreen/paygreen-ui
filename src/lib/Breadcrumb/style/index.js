import styled from 'styled-components';
import { textColor } from './constants';

const BreadcrumbBase = styled.ul`
    margin: ${props => props.theme.space.md} 0;
    padding: 0;
    list-style-type: none;
    display: flex;
    flex-wrap: wrap;
    font-size: ${props => props.theme.font.size.xs};
    font-weight: ${props => props.theme.font.weight.bold};
    text-transform: uppercase;
    letter-spacing: ${props => props.theme.font.spacing};

    li {
        display: flex;
        align-items: center;
        padding: ${props => props.theme.space.xs} 0;

        a {
            outline: none;
            color: ${props => textColor.link[props.colorType]};
            transition: all ${props => props.theme.transition.xs};
        }

        &:not(:last-of-type) {
            a {
                &:hover,
                &:active,
                &:focus {
                    color: ${props => textColor.hover[props.colorType]};
                }
            }
        }

        &:last-of-type {
            a {
                cursor: default;
                color: ${props => textColor.default[props.colorType]};
            }
        }

        .icon {
            opacity: .8;
        }
    }
`;

export { BreadcrumbBase };