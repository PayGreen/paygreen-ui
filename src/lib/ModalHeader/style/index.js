import styled from 'styled-components';
import { responsiveSpaces } from '../../../shared/spaces';

const ModalHeaderBase = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;

    .icon {
        margin-left: auto; // needed to align on the right side the cross if no title in modalHeader
    }

    grid-area: header;
    ${responsiveSpaces('padding')};
`;

export { ModalHeaderBase };
