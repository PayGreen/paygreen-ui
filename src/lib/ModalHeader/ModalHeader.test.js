import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import { ModalHeader } from './style/subComponents';
import Title from '../Title/Title';
import { CrossIcon } from '../Icon/Icon';

it('renders without crashing', () => {
    const modalHeader = TestRenderer.create(
            <ModalHeader theme={ThemeDefault}>
                <Title theme={ThemeDefault}>Que voulez-vous faire ?</Title>

                <CrossIcon theme={ThemeDefault} />
            </ModalHeader>
    );
    expect(modalHeader.toJSON()).toMatchSnapshot();
});
