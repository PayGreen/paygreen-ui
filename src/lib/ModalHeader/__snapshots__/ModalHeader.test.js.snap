// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__ModalHeaderBase-e1z6bj-0 clfhBu"
>
  <span
    className="style__TitleBase-td7fx4-0 hQMgyM"
  >
    This is the Title
  </span>
  <span
    className="style__IconBase-ak3679-0 cZjoZu icon"
  >
    <svg
      viewBox="0 0 143 143"
    >
      <path
        d="M74.6 71.5l43.1-43.1c1.4-1.4 1.4-3.6 0-4.9-1.4-1.4-3.6-1.4-4.9 0L69.6 66.6 26.5 23.5c-1.4-1.4-3.6-1.4-4.9 0-1.4 1.4-1.4 3.6 0 4.9l43.1 43.1-43.1 43.1c-1.4 1.4-1.4 3.6 0 4.9.7.7 1.6 1 2.5 1s1.8-.3 2.5-1l43.1-43.1 43.1 43.1c.7.7 1.6 1 2.5 1s1.8-.3 2.5-1c1.4-1.4 1.4-3.6 0-4.9L74.6 71.5z"
      />
    </svg>
  </span>
</div>
`;
